### PR TITLE
Fix the Add Api to Product and Fix Error model in OperationResultContract

### PR DESCRIPTION
**Pull request by solankisamir**
## Description
This is changes for bug fixes in Spec.
**Issue**: https://github.com/Azure/azure-rest-api-specs/issues/3260
**Swagger Spec changes**:
https://github.com/Azure/azure-rest-api-specs/pull/3258
https://github.com/Azure/azure-rest-api-specs/pull/3272

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/ApiManagement/Management.ApiManagement/Microsoft.Azure.Management.ApiManagement.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <Description>Provides ApiManagement management capabilities for Microsoft Azure.</Description>
     <AssemblyTitle>Microsoft Azure API Management Management</AssemblyTitle>
     <AssemblyName>Microsoft.Azure.Management.ApiManagement</AssemblyName>
-    <Version>4.0.2-preview</Version>
+    <Version>4.0.3-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/ApiManagement/Management.ApiManagement/Microsoft.Azure.Management.ApiManagement.csproj]: Please update the `AssmeblyFileVersion` to `4.0.3` [here](https://github.com/solankisamir/azure-sdk-for-net/blob/242f567665422b0005716f1d138a0ef4faa2cfbb/src/SDKs/ApiManagement/Management.ApiManagement/Properties/AssemblyInfo.cs)

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please create a `generate.ps1` similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Management.Compute/generate.ps1) and regenerate the code.
Also, delete the generate.cmd file

File: src/SDKs/ApiManagement/Management.ApiManagement/Microsoft.Azure.Management.ApiManagement.csproj (hunk 1 above)
Comment (solankisamir) [reply src/SDKs/ApiManagement/Management.ApiManagement/Microsoft.Azure.Management.ApiManagement.csproj]: Fixed

**Comment by solankisamir** [comment]
@dsgouda can you take a look?

**Comment by solankisamir** [comment]
@dsgouda can you take a look?

File: src/SDKs/_metadata/apimanagement_resource-manager.txt [hunk 2]
```diff
@@ -1,11 +1,69 @@
...
+Generating CSharp code
+Executing AutoRest command
+cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/apimanagement/resource-manager/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=D:\github\azure-sdk-for-net\src\SDKs
+AutoRest code generation utility [version: 2.0.4280; node: v8.9.3]
+(C) 2018 Microsoft Corporation.
+https://aka.ms/autorest
+   Loading AutoRest core      'C:\Users\Sasolank\.autorest\@microsoft.azure_autorest-core@2.0.4280\node_modules\@microsoft.azure\autorest-core\dist' (2.0.4280)
+   Loading AutoRest extension '@microsoft.azure/autorest.csharp' (~2.2.51->2.2.57)
+   Loading AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.43->2.3.43)
+WARNING: "HEAD 'GetEntityTag' method missing 404 status code section -- this might be unintentional.
```
Comment (dsgouda) [inline src/SDKs/_metadata/apimanagement_resource-manager.txt]: @fearthecowboy any reason why AutoRest is emitting these warnings?
@shahabhijeet FYI

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**